internal/topology/check/compatibility.go

```diff
@@ -74,9 +74,11 @@ func ObjectsAreInTheSameNamespace(current, desired client.Object) field.ErrorLis
 
 	// NOTE: this should never happen (webhooks prevent it), but checking for extra safety.
 	if current.GetNamespace() != desired.GetNamespace() {
-		allErrs = append(allErrs, field.Forbidden(
+		allErrs = append(allErrs, field.Invalid(
 			field.NewPath("metadata", "namespace"),
-			fmt.Sprintf("cannot be changed from %v to %v to prevent incompatible changes in %v/%v", current.GetNamespace(), desired.GetNamespace(), current.GetObjectKind().GroupVersionKind().GroupKind().String(), current.GetName()),
+			desired.GetNamespace(),
+			fmt.Sprintf("templates must be in the same namespace of the Cluster (%s)",
```

Comment:
```suggestion
			fmt.Sprintf("templates must be in the same namespace as the Cluster (%s)",
```
nit, maybe. If yes, same in l.120 and l.156